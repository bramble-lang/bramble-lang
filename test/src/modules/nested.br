fn my_main() -> i64 {
    let x:i64 := my_mod::inner::add(3, 2);
    root::std::io::writei64ln(x);

    let b:bool := root::my_mod::is_3(1);
    root::std::io::writeboolln(b);

    let b2:bool := root::my_mod::is_3(3);
    root::std::io::writeboolln(b2);
    
    let y:i64 := self::my_mod::inner::add(7, 2);
    root::std::io::writei64ln(y);

    return;
}

mod my_mod {
    fn is_3(i: i64) -> bool {
        let x: i64 := self::inner::add(1, 2);
        return i == x;
    }
    
    mod inner {

        fn add(a: i64, b: i64) -> i64 {
            return a + b;
        }
    }
}