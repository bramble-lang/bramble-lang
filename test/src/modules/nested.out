5
false
true
9
